packages/amplify_auth_cognito/ios/Classes/FlutterFetchCognitoSessionResult.swift

```
@@ -30,70 +31,142 @@ struct FlutterFetchCognitoSessionResult {
+            if case .signedOut = error as! AuthError {
+              // if guest access is possible, return nil userSub
+              if (guest) {
+                return nil
+              } else {
+                throw error as! AuthError
+              }
+            } else {
+              throw error as! AuthError
+            }
+        }
       }
     }
     return sub
   }
 
-  func getIdentityId(session: AuthSession) throws -> String {
-    var id: String = ""
+  func getIdentityId(session: AuthSession) throws -> String? {
+    let userPoolOnly = signedInToUserPoolWithoutCIDP(session: session)
+    var id: String? = nil
+    if (userPoolOnly) {
+      return id
+    }
     if let identityProvider = session as? AuthCognitoIdentityProvider {
       do {
         id = try identityProvider.getIdentityId().get()
       } catch {
-         throw error as! AuthError
+        throw error as! AuthError
       }
     }
     return id
   }
 
-  func getCredentials(session: AuthSession) throws -> [String: String] {
+func getCredentials(session: AuthSession) throws -> [String: String] {
     var credentialMap: [String: String] = [:]
-
+    let userPoolOnly = signedInToUserPoolWithoutCIDP(session: session)
+    if (userPoolOnly) {
+        return credentialMap
+    }
     if let awsCredentialsProvider = session as? AuthAWSCredentialsProvider {
-      let creds =  try awsCredentialsProvider.getAWSCredentials().get()
+      do {
+        let creds =  try awsCredentialsProvider.getAWSCredentials().get()
         credentialMap["awsAccessKey"] = creds.accessKey
         credentialMap["awsSecretKey"] = creds.secretKey
+      } catch {
+        throw error as! AuthError
+      }
+      do {
+        let tempCreds = try awsCredentialsProvider.getAWSCredentials().get() as? AuthAWSTemporaryCredentials
+          if (tempCreds?.sessionKey != nil) {
+            credentialMap["sessionToken"] = tempCreds?.sessionKey
+          }
+      } catch {
+        throw error as! AuthError
```

why do we need to force cast here? Can we simply throw the error?

```swift


         let creds =  try awsCredentialsProvider.getAWSCredentials().get()
         credentialMap["awsAccessKey"] = creds.accessKey
         credentialMap["awsSecretKey"] = creds.secretKey

         let tempCreds = try awsCredentialsProvider.getAWSCredentials().get() as? AuthAWSTemporaryCredentials
           if (tempCreds?.sessionKey != nil) {
             credentialMap["sessionToken"] = tempCreds?.sessionKey
           }
   

```